sqlmesh/core/loader.py

```diff
@@ -379,4 +381,9 @@ def _model_cache_entry_id(self, model_path: Path) -> str:
                 self._loader._config_mtimes.get(self._context_path),
                 self._loader._config_mtimes.get(self._loader._context.sqlmesh_path),
             ]
-            return str(int(max(m for m in mtimes if m is not None)))
+            return "__".join(
+                [
+                    str(int(max(m for m in mtimes if m is not None))),
+                    self._loader._context.config.fingerprint,
```

Comment:
We didn't include the actual fingerprint of the config in this which means if someone modified the config in values they passed into Context then we could reuse cache even if they changed a config that would modify the behavior of models. So this makes it based on provided config and last modified date.

Reply:
Is this something we need in the dbt loader too? Maybe not

Reply:
I would guess not since most would be loading from files which I'm assuming has a modified timestamp detection like what we are doing here.